src/test/kotlin/systems/ajax/malov/stockanalyzer/service/impl/FinnhubMongoStockRecordClientApiTest.kt

```diff
@@ -28,15 +28,34 @@ class FinnhubMongoStockRecordClientApiTest {
 
     @Test
     fun `getAllStockRecords calls Finnhub API and retrieves stock records`() {
+        // GIVEN
         val expected = listOf(unsavedStockRecord())
         val quote = Quote(1f, 1f, 1f, 1f, 1f, 1f, 1f)
-        whenever(finnhubStockApi.quote(TEST_STOCK_SYMBOL))
-            .thenReturn(quote)
+        every {
+            finnhubStockApi.quote(TEST_STOCK_SYMBOL)
+        } returns quote
 
-        val actual = finnhubStockRecordClientApi.getAllStockRecords()
+        // WHEN
+        val actual: Flux<MongoStockRecord> = finnhubStockRecordClientApi.getAllStockRecords()
 
-        verify(finnhubStockApi).quote(eq(TEST_STOCK_SYMBOL))
-        assertEquals(expected.size, actual.size)
-        assertEquals(expected[0].symbol, actual[0].symbol)
+        // THEN
+        StepVerifier.create(actual)
+            .expectNextMatches { it.symbol == expected[0].symbol }
+            .verifyComplete()
+    }
+
+    @Test
+    fun `getAllStockRecords returns empty flux when client throws exception`() {
```

Comment:
why we receive empty flux instead of error?

Reply:
In this case it's fine to get an error from external client. Third-party can be just busy and we get 429 status code